nixos/modules/services/web-apps/pict-rs.nix

```diff
@@ -0,0 +1,45 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let
+  cfg = config.services.pict-rs;
+in
+{
+  options.services.pict-rs = {
+    enable = mkEnableOption "pict-rs server";
+    imageDirectory = mkOption {
+      type = types.path;
+      default = "/var/lib/pict-rs";
+      description = ''
+        The directory where to store the uploaded images.
+      '';
+    };
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        The IPv4 address to deploy the service to.
+      '';
+    };
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      description = ''
+        The port which to bind the service to.
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
```

Comment:
Even more bonus points of this gets a NixOS test. I think a test would be helpful.

Reply:
you're right!